docs/tutorials/gallery_adding_components.py

```diff
@@ -0,0 +1,311 @@
+
+# %%
+# We next import the [ragna.Rag][] class and set up a chat using the custom objects from above
+
+from ragna import Rag
+
+chat = Rag().chat(
+    documents=[document_path],
+    source_storage=TutorialSourceStorage,
+    assistant=TutorialAssistant,
+)
+
+# %%
+# Before we can ask a question, we need to [`prepare`][ragna.core.Chat.prepare] the chat, which
+# under the hood stores the documents we have selected in the source storage.
+
+_ = await chat.prepare()
+
+# %%
+# Finally, we can get an [`answer`][ragna.core.Chat.answer] to a question.
+
+print(await chat.answer("What is Ragna?"))
+
+# %%
+# ## Using the Web UI with Custom Objects
+
+# %%
+# To include our custom objects in the Web UI, make sure they are in your
+# [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH),
+# and list them in your [configuration file](../../references/config.md).
+
+# %%
+# Here is an example configuration file in the case that you would like to use one:
+
+# %%
+# ```toml
+# local_root = "/home/ragnauser/.cache/ragna"
+# authentication = "ragna.deploy.RagnaDemoAuthentication"
+# document = "ragna.core.LocalDocument"
+# source_storages = [
+#     "tutorial.TutorialSourceStorage"
+# ]
+# assistants = [
+#     "tutorial.TutorialAssistant"
+# ]
+#
+# [api]
+# hostname = "127.0.0.1"
+# port = 31476
+# root_path = ""
+# url = "http://127.0.0.1:31476"
+# database_url = "sqlite:////home/ragnauser/.cache/ragna/ragna.db"
+# origins = [
+#     "http://127.0.0.1:31477",
+# ]
+#
+# [ui]
+# hostname = "127.0.0.1"
+# port = 31477
+# origins = [
+#     "http://127.0.0.1:31477",
+# ]
+# ```
+
+# %%
+# Note how the `tutorial.TutorialSourceStorage` and `tutorial.TutorialAssistant` classes are listed
+# in the `source_storages` and `assistants` fields, respectively.
+
+
+# %%
+# ## Using the REST API with Custom Objects
+
+# %%
+# This section will follow the steps of
+# [the REST API tutorial](../../generated/tutorials/gallery_rest_api.md). More detail
+# can be found there. This section focuses specifically on using custom objects.
+
+# %%
+# We first set up the [configuration](../../references/config.md) and start the API
+
+import sys
+from pathlib import Path
+
+sys.path.insert(0, str(Path.cwd().parent))
+
+import documentation_helpers
+
+from ragna.deploy import Config
+
+import httpx
+
+config = Config()
+
+config.source_storages = [TutorialSourceStorage]
+config.assistants = [TutorialAssistant]
+
+rest_api = documentation_helpers.RestApi()
+_ = rest_api.start(config)
+
+# %%
+# Next, let's authenticate ourselves.
+
+username = password = "Ragna"
+
+client = httpx.Client(base_url=config.api.url)
+response = client.post(
+    "/token",
+    data={"username": username, "password": password},
+).raise_for_status()
+token = response.json()
+
+# %%
+# We set the API token on our HTTP client so we don't have to manually supply it
+# for each request below.
+
+client.headers["Authorization"] = f"Bearer {token}"
```

Comment:
This will give you a fully authenticated client

```suggestion
client = rest_api.start(config, authenticate=True)
```